src/sparseml/transformers/sparsification/trainer.py

```diff
@@ -444,52 +421,29 @@ def _setup_manager(
         self, kwargs
     ) -> Tuple[Optional[ScheduledModifierManager], List[ScheduledModifierManager]]:
         manager = None
-        arch_managers = []
-
+        checkpoint_recipe = os.path.join(self.model_state_path, RECIPE_NAME)
+        # By default we should fetch checkpoint_recipe from self.model_state_path
+        # This is used for testing only
+        # checkpoint_recipe = os.path.join(
+        #    kwargs["args"].to_dict()["output_dir"], RECIPE_NAME
+        # )
         if self.recipe is not None:
             manager = ScheduledModifierManager.from_yaml(
                 self.recipe,
                 recipe_variables=self.recipe_args,
                 metadata=self.metadata,
             )
+        if os.path.isfile(checkpoint_recipe) and manager:
+            manager = ScheduledModifierManager.compose_staged(
+                base_recipe=checkpoint_recipe, additional_recipe=manager
+            )
             _LOGGER.info(
                 "Loaded SparseML recipe variable into manager for recipe: "
                 f"{self.recipe}, recipe_variables: {self.recipe_args} "
                 f"and metadata {self.metadata}"
             )
 
-        arch_recipe_paths = glob.glob(os.path.join(self.model_state_path, RECIPE_REGEX))
```

Comment:
I think we can get rid of RECIPE_REGEX and stuff around that now, but please double check